helix-core/src/main/java/org/apache/helix/controller/stages/ClusterEventType.java

```diff
@@ -37,6 +37,7 @@
   Resume,
   PeriodicalRebalance,
   OnDemandRebalance,
+  ControllerChange,
```

Comment:
I think using OnDemandRebalance is good enough. ControllerChange could be misleading since not all controller change notifications are generating an event. It is actually mainly processed by the onControllerChange(). The event is a derivation from the real processing logic.

Reply:
Synced with Huizhi offline. So this type will be a general event type to trigger entering special mode or exit special mode. It is potentially more handy than Resume and has a different meaning of OnDemandRebalance. He convinced me that it would be a good idea to add it.